plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,230 @@
+#!powershell
+
+# Copyright: (c) 2020, Jamie Magee <jamie.magee@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.ArgvParser
+#Requires -Module Ansible.ModuleUtils.CommandUtil
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+  options = @{
+    arch        = @{ type = "str"; choices = "32bit", "64bit" }
+    independent = @{type = "bool"; default = $false }
+    global      = @{ type = "bool"; default = $false }
+    name        = @{ type = "list"; elements = "str"; required = $true }
+    no_cache    = @{type = "bool"; default = $false }
+    purge       = @{type = "bool"; default = $false }
+    skip        = @{type = "bool"; default = $false }
+    state       = @{ type = "str"; default = "present"; choices = "present", "absent" }
+  }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$arch = $module.Params.arch
+$independent = $module.Params.independent
+$global = $module.Params.global
+$name = $module.Params.name
+$no_cache = $module.Params.no_cache
+$purge = $module.Params.purge
+$skip = $module.Params.skip
+$state = $module.Params.state
+
+$module.Result.rc = 0
+
+function Install-Scoop {
+
+  # Scoop doesn't have refreshenv like Chocolatey
+  # Let's try to update PATH first
+  $env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")
```

Comment:
Why is this needed, by default PowerShell will have the Machine, User, and Process scoped env vars under the `env:` provider.

Reply:
It's a bit of a hack. Scoop doesn't update the current session with environment variable changes. Without this line, or rather without the change on line 79, scoop isn't found in the path. So even after scoop is installed here, it's not found for any of the subsequent commands.

I know this isn't the best, but I am open to alternatives

Reply:
I suppose this is simpler then just trying to add the scoop path to `$env:PATH` as it will get whatever changes the scoop installer adds.

Reply:
I'm resolving this for now, but open to any alternatives.